eng/common/scripts/Helpers/DevOps-WorkItem-Helpers.ps1

```diff
@@ -1015,3 +1015,64 @@ function UpdateValidationStatus($pkgvalidationDetails, $BuildDefinition, $Pipeli
     Write-Host "[$($workItem.id)]$LanguageDisplayName - $pkgName($versionMajorMinor) - Updated"
     return $true
 }
+
+function Get-ReleasePlansForCPEXAttestation()
+{
+  $fields = @()
+  $fields += "Custom.ProductServiceTreeID"
+  $fields += "Custom.ReleasePlanType"
+  $fields += "Custom.ProductType"
+  $fields += "Custom.DataScope"
+  $fields += "Custom.MgmtScope"
+
+  $fieldList = ($fields | ForEach-Object { "[$_]"}) -join ", "
+  $query = "SELECT ${fieldList} FROM WorkItems WHERE [System.WorkItemType] = 'Release Plan' AND [System.State] = 'Finished'"
+  $query += " AND [Custom.AttestationStatus] IN ('', 'Pending')"
+  $query += " AND [System.Tags] NOT CONTAINS 'Release Planner App Test'"
+  $query += " AND [System.Tags] NOT CONTAINS 'Release Planner Test App'"
+  $query += " AND [System.ChangedDate] >= @StartOfDay('-1')" # Filter for items changed yesterday only
+  $query += " AND [System.ChangedDate] < @StartOfDay('0')"
```

Comment:
Why do we need this condition? This will filter out any work items modified on Friday after attestation job triggered and if job does not run on weekends. Similarly let's say, job fails for couple of days for some external dependency reasons then it will skip all pending items from previous days when attestation job resumes. We should pick up all finished release plans in pending or empty attestation status.